src/ccnl-core/src/ccnl-relay.c

```diff
@@ -556,8 +557,13 @@ ccnl_content_add2cache(struct ccnl_relay_s *ccnl, struct ccnl_content_s *c)
         }
     }
 
+#ifdef CCNL_RIOT
+    if (ccnl->max_cache_entries > 0 &&
+        ccnl->contentcnt >= ccnl->max_cache_entries && !cache_strategy_remove(ccnl, c, tclass)) {
```

Comment:
If no replacement strategy is implemented, `cache_strategy_remove()` returns 0 and the default replacement strategy (defined below this line) is applied. If the specified replacement strategy decides not to replace any content (e.g. due to QoS constraints), it should still return 1 to prevent the default replacement from being applied.